packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -646,43 +675,157 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
+  ///             mainAxisAlignment: MainAxisAlignment.spaceAround,
+  ///             children: <Widget>[
+  ///               ...List<Widget>.generate(UnfocusDisposition.values.length,
+  ///                   (int index) {
+  ///                 return Row(
+  ///                   mainAxisSize: MainAxisSize.min,
+  ///                   children: <Widget>[
+  ///                     Radio<UnfocusDisposition>(
+  ///                       groupValue: disposition,
+  ///                       onChanged: (UnfocusDisposition value) {
+  ///                         setState(() {
+  ///                           disposition = value;
+  ///                         });
+  ///                       },
+  ///                       value: UnfocusDisposition.values[index],
+  ///                     ),
+  ///                     Text(describeEnum(UnfocusDisposition.values[index])),
+  ///                   ],
+  ///                 );
+  ///               }),
+  ///               OutlineButton(
+  ///                 child: const Text('UNFOCUS'),
+  ///                 onPressed: () {
+  ///                   setState(() {
+  ///                     primaryFocus.unfocus(disposition: disposition);
+  ///                   });
+  ///                 },
+  ///               ),
+  ///             ],
+  ///           ),
+  ///         ],
+  ///       ),
+  ///     ),
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
+  void unfocus({
+    UnfocusDisposition disposition = UnfocusDisposition.scope,
+  }) {
+    assert(disposition != null);
     if (!hasFocus && (_manager == null || _manager._markedForFocus != this)) {
       return;
     }
-    if (!hasPrimaryFocus) {
-      // If we are in the focus chain, but not the primary focus, then unfocus
-      // the primary instead.
-      _manager?.primaryFocus?.unfocus(focusPrevious: focusPrevious);
+    FocusScopeNode scope = enclosingScope;
+    if (scope == null) {
+      // If the scope is null, then this is either the root node, or a node that
+      // is not yet in the tree, neither of which do anything when unfocused.
+      return;
     }
-    _manager?._markUnfocused(this);
-    final FocusScopeNode scope = enclosingScope;
-    if (scope != null) {
-      scope._focusedChildren.remove(this);
-      if (focusPrevious) {
-        scope._doRequestFocus();
-      }
+    switch (disposition) {
+      case UnfocusDisposition.scope:
+        // If it can't request focus, then don't modify its focused children.
+        if (scope.canRequestFocus) {
+          // Clearing the focused children here prevents re-focusing the node
+          // that we just unfocused if we immediately hit "next" after
+          // unfocusing, and also prevents choosing to refocus the next-to-last
+          // focused child if unfocus is called more than once.
+          scope._focusedChildren.clear();
+        }
+
+        while (!scope.canRequestFocus) {
+          scope = scope.enclosingScope ?? _manager?.rootScope;
```

Comment:
I am on the fence about clearing anything though: it might make more sense to just focus the scope, and not touch the `_focusedChildren` here. It would mean that as soon as you hit "next", you get back to the thing you just unfocused, but I'm not sure that's wrong.

Reply:
thanks for explaining. I think there might not be a correct answer, it is down to different use cases. I am ok with keep what we have now and maybe tweak it later if needed

Reply:
Yes, part of why I made this into an enum is so we can add other modes later without breaking things. And we can tweak it if it turns out to just be wrong.